tests/functional/TestErrorHandling/tests/timeout_test.go

```diff
@@ -0,0 +1,124 @@
+// +build functional
+
+//
+// Copyright (c) 2019 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+package tests
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	rex "github.com/heketi/heketi/pkg/remoteexec"
+	"github.com/heketi/heketi/pkg/remoteexec/ssh"
+	"github.com/heketi/heketi/pkg/testutils"
+	"github.com/heketi/heketi/server/config"
+
+	"github.com/heketi/tests"
+)
+
+var (
+	pidFile   = "/var/tmp/hook.pid"
+	delayFile = "/var/tmp/hook-delay"
+	killPid   = "/var/tmp/kill-pid.sh"
+)
+
+func addDelay(timeOut uint16, node string, exec *ssh.SshExec) error {
+	dumpData := fmt.Sprintf("echo -e %dm >%s", timeOut, delayFile)
+	cmd := []string{dumpData}
+	err := rex.AnyError(exec.ExecCommands(node, cmd, 10, true))
+	return err
+}
+
+func cleanDelay(node string, exec *ssh.SshExec) error {
+	removeSleep := fmt.Sprintf("sh %s", killPid)
+	removeDelayFile := fmt.Sprintf("rm -f %s", delayFile)
+	removePIDFile := fmt.Sprintf("rm -f %s", pidFile)
+
+	cmd := []string{removeSleep, removeDelayFile, removePIDFile}
+	err := rex.AnyError(exec.ExecCommands(node, cmd, 10, true))
+	return err
+}
+
+func cleanupOp(exec *ssh.SshExec, t *testing.T) {
+	for i := 0; i < len(testCluster.Nodes); i++ {
+		sshHost := testCluster.SshHost(i)
+		err := cleanDelay(sshHost, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	}
+}
+
+func TestGlusterTimeout(t *testing.T) {
+
+	heketiServer := testutils.NewServerCtlFromEnv("..")
+	origConf := path.Join(heketiServer.ServerDir, heketiServer.ConfPath)
+	heketiServer.ConfPath = tests.Tempfile()
+	defer os.Remove(heketiServer.ConfPath)
+	err := UpdateConfig(origConf, heketiServer.ConfPath, func(c *config.Config) {
+		c.GlusterFS.RetryLimits.VolumeCreate = 1
+		//gluster cli timeout set to 4 minutes
+		c.GlusterFS.SshConfig.GlusterCliTimeout = 240
+	})
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	err = heketiServer.Start()
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	defer heketiServer.Stop()
+
+	testCluster.Teardown(t)
+	testCluster.Setup(t, len(testCluster.Nodes), len(testCluster.Disks))
+	defer testCluster.Teardown(t)
+
+	na := testutils.RequireNodeAccess(t)
+	exec := na.Use(logger)
+	for i := 0; i < len(testCluster.Nodes); i++ {
+
+		//the volume we are creating is replica 3
+		//the bricks get created on 3 nodes the script
+		//will also get executed on 3 nodes.
+		//1+1+1 =3 minutes to ensure we are not hitting
```

Comment:
I think this still needs discussion. I will try to talk to you online sometime about it.

Reply:
sure